Source/Internal/IGListPreprocessor.h

```diff
@@ -0,0 +1,34 @@
+//
+//  IGListPreprocessor.h
+//  IGListKit
+//
+//  Created by Adlai Holler on 2/20/17.
+//  Copyright © 2017 Instagram. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+#import <IGListKit/IGListMacros.h>
+#import <IGListKit/IGListPreprocessingContext.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class IGListSectionController, IGListSectionMap;
+
+IGLK_SUBCLASSING_RESTRICTED
+@interface IGListPreprocessor : NSObject
+
+- (instancetype)initWithSectionMap:(IGListSectionMap *)sectionMap;
+
+/**
+ * Must be called from the main thread.
+ */
+- (void)schedulePreprocessingWithContainerSize:(CGSize)containerSize completion:(dispatch_block_t)completionBlock;
```

Comment:
If size changes and your preprocessing uses the size changes, you'll have to kick off updates again. You could call `adapter.performUpdates(...)` with each bounds change (it'll coalesce) to do that. Atm IGListKit doesn't do much w/ layout (aside from the provided collection view layout).

> does there need to be a way to cancel what may be long running background processes that are no longer valid

Maybe. Ideally nothing should be that long running: imagine just calculating text sizes for caches.

The only other to do this would be to lock on the [containerSize](https://github.com/Instagram/IGListKit/blob/master/Source/IGListAdapter.m#L810) API, but that wont even solve racing since the value could change between begin/end of processing.